lib/Doctrine/DBAL/Schema/Comparator.php

```diff
@@ -426,7 +427,11 @@ public function diffColumn(Column $column1, Column $column2)
 
         $changedProperties = [];
 
-        if (get_class($properties1['type']) !== get_class($properties2['type'])) {
+        // Exception here helps with specific case of base string-type and a custom enum-type (inheriting from string).
+        // When doing migration diffs, there is no need to flag the column as different (unless options differ too)
+        $typeClass1 = get_class($properties1['type']);
+        $typeClass2 = get_class($properties2['type']);
+        if ($typeClass1 !== $typeClass2 && ! $this->isStringParentOf($typeClass1, $typeClass2)) {
```

Comment:
Some types require a comment hint:

https://github.com/doctrine/dbal/blob/dbc623d40ad18c24274735c1c85fdaa1dee3323f/lib/Doctrine/DBAL/Types/ArrayType.php#L69-L72

Even if the two types are implemented by a child and the parent classes, they may produce different DDL and require a migration.

Reply:
Hi all. Thank you for taking time to look at this PR and sorry for the very late response!

In hindsight, I think this change was a bit naive and too niche to be part of the core mechanism as you have suggested. The tricky part for me is that the dbal migration code is really closed off (no dependency injection so I can't really get at the code in any sane way) and pretty much impossible to modify in the way I need above in user-land. I think my options are to:
1. - Just add the comments to the enum columns to help dbal out (and the risks that dual migration managers entail). This may not be too bad but I envision issues when the current migration system e.g., removes/renames a column
2. - Parse and remove unwanted enum/string diffs in a post process
3. - Continue to maintain a custom fork until we can switch the migrations source of truth over to Doctrine

I'm prob gonna go with the latter..

Happy to close this PR. Thanks again for your time